File: 4. Reverse Array Using Stack - GFG/4.-reverse-array-using-stack.cpp
// { Driver Code Starts
#include<bits/stdc++.h>
using namespace std;



 // } Driver Code Ends
//User function Template for C++

//Complete this function
class Solution{
  public:
    void reverseArray(int n, int* arr){
        
      //Your code here
      stack<int> st;
      for(int i = 0; i<n; i++){
          st.push(arr[i]);
      }
      
      //Reverse the array
      int i = 0;
      while(!st.empty()){
          arr[i] = st.top();
          st.pop();
          i++;
      }
    }
};

// { Driver Code Starts.
int main(){
    int t;
    cin>>t;
    while(t--)
    {
    int n;
    cin >> n;
    int arr[n];
    for(int i = 0;i<n;i++){
        cin >> arr[i];
    }
    Solution obj;
    obj.reverseArray(n, arr);
    
    for(int i = 0;i<n;i++){
        cout << arr[i] << " ";
    }
    
    cout << endl;
    }
    return 0;
}  // } Driver Code Ends

File: STL ( Standard_Template_library )/Stack/4. Reverse Array Using Stack - GFG/4.-reverse-array-using-stack.py
#User function Template for python3

class Solution:
    def reverseArray(self,n,arr):
        #code here
        for i in range(n//2):
            arr[i],arr[-(1+i)] = arr[-(1+i)],arr[i]

#{ 
#  Driver Code Starts
#Initial Template for Python 3

#contributed by RavinderSinghPB
if __name__=='__main__':
    tcs=int(input())

    for _ in range(tcs):
        n=int(input())
        arr=[int(x) for x in input().split()]

        Solution().reverseArray(n, arr)

        for e in arr:
            print(e,end=' ')
        print()
# } Driver Code Ends